misk/src/test/kotlin/misk/client/ProtoMessageHttpClientTest.kt

```diff
@@ -17,53 +22,79 @@ import misk.web.actions.WebAction
 import misk.web.jetty.JettyService
 import misk.web.mediatype.MediaTypes
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import javax.inject.Inject
+import javax.inject.Singleton
 
 @MiskTest(startService = true)
 class ProtoMessageHttpClientTest {
-  @MiskTestModule
-  val module = Modules.combine(
-      MiskModule(),
-      WebModule(),
-      TestWebModule(),
-      TestModule()
-  )
+    @MiskTestModule
+    val module = Modules.combine(
+            MiskModule(),
+            WebModule(),
+            TestWebModule(),
+            TestModule()
+    )
 
-  private @Inject lateinit var jettyService: JettyService
-  private @Inject lateinit var httpClient: ProtoMessageHttpClient
+    @Inject
+    private lateinit var jetty: JettyService
 
-  @Test
-  fun protoMessageHttpCall() {
-    val dinoMessage = Dinosaur.Builder()
-        .name("stegosaurus")
-        .picture_urls(listOf(
-            "https://cdn.dinopics.com/stego.jpg",
-            "https://cdn.dinopics.com/stego2.png"
+    private lateinit var httpClient: ProtoMessageHttpClient
+
+    @BeforeEach
+    fun createClient() {
+        val clientInjector = Guice.createInjector(MoshiModule(), ClientModule(jetty))
+        httpClient = clientInjector.getInstance(Names.named("dinosaur"))
+    }
+
+    @Test
+    fun protoMessageHttpCall() {
+        val dinoMessage = Dinosaur.Builder()
+                .name("stegosaurus")
+                .picture_urls(listOf(
+                        "https://cdn.dinopics.com/stego.jpg",
+                        "https://cdn.dinopics.com/stego2.png"
+                ))
+                .build()
+
+        val response = httpClient.post<Dinosaur>("/cooldinos", dinoMessage)
+        assertThat(response.name).isEqualTo("supersaurus")
+        assertThat(response.picture_urls).isEqualTo(listOf(
+                "https://cdn.dinopics.com/stego.jpg",
+                "https://cdn.dinopics.com/stego2.png"
         ))
-        .build()
+    }
+
+    class ReturnADinosaur : WebAction {
+        @Post("/cooldinos")
+        @RequestContentType(MediaTypes.APPLICATION_JSON)
+        @ResponseContentType(MediaTypes.APPLICATION_JSON)
+        fun getDinosaur(@RequestBody requestBody: Dinosaur):
+                Dinosaur = requestBody.newBuilder().name("supersaurus").build()
+    }
+
+    class TestModule : KAbstractModule() {
+        override fun configure() {
+            install(WebActionModule.create<ReturnADinosaur>())
+        }
+    }
 
-    val response = httpClient.post<Dinosaur>(
-        jettyService.serverUrl.toString(), "/cooldinos", dinoMessage)
-    assertThat(response.name).isEqualTo("supersaurus")
-    assertThat(response.picture_urls).isEqualTo(listOf(
-        "https://cdn.dinopics.com/stego.jpg",
-        "https://cdn.dinopics.com/stego2.png"
-    ))
-  }
+    // NB(mmihic): The server doesn't get a port until after it starts, so we
```

Comment:
hmmmm . . . but we _do_ know it from configs right?

(not that we know the hostname)